src/carbonio-files-ui-common/views/components/CopyNodesModalContent.test.tsx

```diff
@@ -155,93 +156,111 @@ describe('Copy Nodes Modal', () => {
-		expect(fileItem).toHaveAttribute('disabled');
-		// folder with write file permission is active and navigable
-		expect(folderWithWriteFileItem).toBeVisible();
-		expect(folderWithWriteFileItem).not.toHaveAttribute('disabled');
-		await user.dblClick(folderWithWriteFileItem);
-		// navigate to sub-folder
-		await screen.findByText((folderWithWriteFile.children.nodes[0] as File | Folder).name);
-		expect(folderWithWriteFileItem).not.toBeInTheDocument();
-
-		// then copy folder
-		nodesToCopy = [folder];
-		rerender(
-			<div onClick={resetToDefault}>
-				<CopyNodesModalContent folderId={currentFolder.id} nodesToCopy={nodesToCopy} />
-			</div>
-		);
-		await screen.findByText((currentFolder.children.nodes[0] as File | Folder).name);
-		folderWithWriteFolderItem = screen.getByText(folderWithWriteFolder.name);
-		folderWithWriteFileItem = screen.getByText(folderWithWriteFile.name);
-		fileItem = screen.getByText(file.name);
-		folderItem = screen.getByText(folder.name);
-		// folder without write folder permission is disabled and not navigable
-		expect(folderWithWriteFileItem).toHaveAttribute('disabled');
-		// double-click on a disabled folder does nothing
-		await user.dblClick(folderWithWriteFileItem);
-		await waitFor(() => expect(folderWithWriteFileItem).toBeVisible());
-		expect(folderWithWriteFileItem).toHaveAttribute('disabled');
-		// moving folder is disabled
-		expect(folderItem).toHaveAttribute('disabled');
-		// file is disabled
-		expect(fileItem).toHaveAttribute('disabled');
-		// folder with write_folder permission is active and navigable
-		expect(folderWithWriteFolderItem).toBeVisible();
-		expect(folderWithWriteFolderItem).not.toHaveAttribute('disabled');
-		await user.dblClick(folderWithWriteFolderItem);
-		// navigate to sub-folder
-		await screen.findByText((folderWithWriteFolder.children.nodes[0] as File | Folder).name);
-		expect(folderWithWriteFolderItem).not.toBeInTheDocument();
-	});
+	describe.each<NonNullable<Node['__typename']>>(['File', 'Folder'])(
+		'when copying a %s',
+		(typename) => {
+			const nodeToCopy = populateNode(typename);
+			nodeToCopy.permissions.can_write_file = true;
+			nodeToCopy.permissions.can_write_folder = true;
+
+			test('files are disabled in the list', async () => {
+				const currentFolder = populateFolder();
+				currentFolder.children.nodes.push(nodeToCopy);
+				const file = populateFile();
+				file.permissions.can_write_file = true;
+				currentFolder.children.nodes.push(file);
+
+				const mocks = [
+					mockGetPath({ node_id: currentFolder.id }, [currentFolder]),
+					mockGetChildren(getChildrenVariables(currentFolder.id), currentFolder)
+				];
+				setup(
+					<div onClick={resetToDefault}>
+						<CopyNodesModalContent folderId={currentFolder.id} nodesToCopy={[nodeToCopy]} />
+					</div>,
+					{ mocks }
+				);
+				await screen.findByText(file.name);
+				const nodeItem = screen.getByText(file.name);
+				expect(nodeItem).toHaveAttribute('disabled', '');
```

Comment:
Unfortunately we can only use toBeDisabled and toBeEnabled on a subset of elements (theoretically, on those elements which can be disabled by the html specification).
In our case, nodeItem is a div, and has the disabled attribute only because it is propagated to the html element by the ds and styled-components.
You can see in the jest-dom library how toBeDisabled and toBeEnabled work, the code is very clear https://github.com/testing-library/jest-dom/blob/main/src/to-be-disabled.js

For sure in the future we will need to replace these kind of checks with something more appropriate

Reply:
thanks, I get it now.